### PR TITLE
Implement base repository

### PR DESCRIPTION
**Pull request by Serge1212**
Approximate time spent: 30 minutes

### CODE REVIEW

File: VetClinic.Core/Interfaces/Repositories/Base/IRepository.cs [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
+namespace VetClinic.Core.Interfaces.Repositories.Base
 {
     public interface IRepository<TEntity>
         where TEntity : class
     {
         Task<IList<TEntity>> GetAsync(
             Expression<Func<TEntity, bool>> filter = null,
             Func<IQueryable<TEntity>, IOrderedQueryable<TEntity>> orderBy = null,
-            string includeProperties = "",
+            Func<IQueryable<TEntity>, IIncludableQueryable<TEntity, object>> include = null,
```
Comment (Serafinium) [inline VetClinic.Core/Interfaces/Repositories/Base/IRepository.cs:16]: Please move all parameters to diff rows

File: VetClinic.DAL/Context/FakeDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace VetClinic.DAL.Context
+{
+    public class FakeDbContext : DbContext
```
Comment (Serafinium) [inline VetClinic.DAL/Context/FakeDbContext.cs:5]: Why is it FakeDbContext ?

File: VetClinic.DAL/Repositories/Base/Repository.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            if(orderBy != null)
+            {
+                return await orderBy(query).ToListAsync();
+            }
+            return await query.ToListAsync();
+        }
+
+        public async Task<TEntity> GetByIdAsync(object id)
+        {
+            return await _context.Set<TEntity>().FindAsync(id);
```
Comment (Serafinium) [inline VetClinic.DAL/Repositories/Base/Repository.cs:49]: What happened if no find?

File: VetClinic.DAL/Context/FakeDbContext.cs (hunk 2 above)
Comment (Serge1212) [reply VetClinic.DAL/Context/FakeDbContext.cs:5]: > Why is it FakeDbContext ?

Just needed some sort of stub in order to implement the base interface that uses DbContext

File: VetClinic.DAL/Repositories/Base/Repository.cs (hunk 3 above)
Comment (Serge1212) [reply VetClinic.DAL/Repositories/Base/Repository.cs:49]: > What happened if no find?

been implementing this method thinking that even tho it doesn't return any element it is a responsibility of controller to check if there is any element found (then return the according status code: not found, ok, etc)

**Pull request merged**